pkgs/by-name/ed/edopro/package.nix

```diff
@@ -0,0 +1,262 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, premake5
+, writeShellScriptBin
+, runCommandLocal
+, symlinkJoin
+, imagemagick
+, bzip2
+, curl
+, flac
+, fmt_9
+, freetype
+, libevent
+, libgit2
+, libGL
+, libGLU
+, libjpeg
+, libpng
+, libvorbis
+, libX11
+, libxkbcommon
+, libXxf86vm
+, lua5_3
+, nlohmann_json
+, noto-fonts-cjk-sans
+, openal
+, SDL2
+, sqlite
+, wayland
+, egl-wayland
+,
+}:
+let
+  edopro-version = "40.1.4";
+
+  # the assets repo has a font instead
+  font = "${noto-fonts-cjk-sans}/share/fonts/opentype/noto-cjk/NotoSansCJK-VF.otf.ttc";
+
+  covers_url = "";
```

Comment:
If we're going through the effort of exposing these variables, might as well add them to the package args? Ditto for the baked-in backup font.

Reply:
What happens if we don't pass in a font then? for the url options the program handles empty strings so those are safe. But I'm unsure how this could be handled.

Reply:
I added it, but now we need to figure out how to inject the build flag if we use a font.

Reply:
```nix
${lib.optionals font "--bundled-font=${font}"}
```
?

Reply:
That's one way, but since the variable has a default value it should just work if you interpolate it.

Alternatively we could build the build args from a list and insert it with `concatStringsSep`, might be nicer anyway.

Reply:
I can't get it to work correctly, I tried to change font to a boolean because thats what lib.optionals wants but it doesn't work.

Reply:
I just removed the font for now, if it causes problems we can just add it back.